lib/mix/ecto.ex

```diff
@@ -97,6 +97,19 @@ defmodule Mix.Ecto do
   end
 
   @doc """
+  Ensures the given repository's migrations path exists on the filesystem.
+  """
+  @spec ensure_migrations_path(Ecto.Repo.t) :: Ecto.Repo.t | no_return
+  def ensure_migrations_path(repo) do
+    path = Path.relative_to(migrations_path(repo), Mix.Project.app_path)
```

Comment:
@josevalim thanks! That was my first thought as well, but I noticed that (within the tests, at least) `migrations_path` was returning `_build/shared/lib/ecto/tmp/Ecto.Migrate/migrations`, instead of just `tmp/Ecto.Migrate/migrations` like I expected.

I believe this is due to [this line](https://github.com/elixir-lang/ecto/blob/master/lib/mix/ecto.ex#L113). I noticed that gen.migration [uses](https://github.com/elixir-lang/ecto/blob/master/lib/mix/tasks/ecto.gen.migration.ex#L43) Mix.Project.app_path, so I thought that would be the way to proceed.

I'm happy to make any adjustments, but I could use some guidance on the correct path.



Application.app_dir